app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -447,6 +449,18 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
+    private fun mapUserType(userType: UserType): Membership {
+        return when (userType) {
+            UserType.GUEST -> Membership.Guest
+            UserType.FEDERATED -> Membership.Federated
+            UserType.EXTERNAL -> Membership.External
+            UserType.INTERNAL -> Membership.None
+            else -> {
+                throw IllegalStateException("Unknown UserType")
+            }
+        }
+    }
+
```

Comment:
Couldn't we have this in a `ConversationMapper`?

Reply:
membership is kind of a. UI model do not think Kalium should know about it ? I added a UserTypeMapper to the AR 

I think in general it is good idea to use mapper classes inside the AR :)